### ISSUE TITLE
Inference gives black output

### ISSUE THREAD

**Issue by dm0288** [aurora95/Keras-FCN#87]
Hi @aurora95  @ahundt ,

I have used the train and inference script along with UNet architecture instead of FCN with PAscal VOC dataset(without augmentation). I was able to train the model with losses going down till 100 epochs. The inference from the model is all black. However, strangely the model from 10 epochs is able to generate result but the class labels are not correct though the region obtained is pretty nice with some erroneous region as well. Could you please guide me where is it going wrong?

Result form 100 Epochs:
![2007_001724](https://user-images.githubusercontent.com/7372398/222345466-9a885c65-f004-4ab2-a725-ed22d81917eb.jpg)
![2007_001289](https://user-images.githubusercontent.com/7372398/222345470-ee851eed-1c3d-43ac-869d-c7f5d832a638.jpg)
Results from 10 EPochs
![2007_001289](https://user-images.githubusercontent.com/7372398/222345234-dd1018d6-0df7-45eb-a0db-b9cda54de059.png)
![2007_001724](https://user-images.githubusercontent.com/7372398/222345245-a0299795-f1b5-45fd-9cf7-a6b5004b7481.png)
